### PR TITLE
ID-29 Email and reset password v2

### PR DESCRIPTION
**Pull request by VishikBhalla**
Added reset password functionality and implemented reset password email limiting. The Email Util class still needs the host name and id to be fully functional.

### CODE REVIEW

File: api/src/main/java/com/codeforcommunity/rest/ApiRouter.java [hunk 1]
```diff
@@ -1,7 +1,7 @@
 package com.codeforcommunity.rest;
 
 //import java.awt.RenderingHints.Key;
-import java.sql.Date;
```
Comment (rymaju) [inline api/src/main/java/com/codeforcommunity/rest/ApiRouter.java:4]: Lets put this back in and prefix everywhere that we used it with sql.Date

File: api/src/main/java/com/codeforcommunity/rest/ApiRouter.java [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
+import java.util.Date;
```
Comment (rymaju) [inline api/src/main/java/com/codeforcommunity/rest/ApiRouter.java:4]: Same with this, we will just prefix where you used util.Date

File: api/src/main/java/com/codeforcommunity/rest/ApiRouter.java [hunk 3]
```diff
@@ -897,6 +898,58 @@ private void handleDeleteNews(RoutingContext ctx) {
       response.setStatusCode(400).end();
     }
   }
+  
+  private void handleResetPasswordEmail (RoutingContext ctx) {
+    long DAY = 24 * 60 * 60 * 1000;
+    HashMap<String, Date> validToken = new HashMap<String, Date>();
```
Comment (rymaju) [inline api/src/main/java/com/codeforcommunity/rest/ApiRouter.java:904]: use util.Date class

File: api/src/main/java/com/codeforcommunity/rest/ApiRouter.java [hunk 4]
```diff
@@ -897,6 +898,58 @@ private void handleDeleteNews(RoutingContext ctx) {
...
+
+    if(!email.endsWith("@husky.neu.edu")) {
+      response.setStatusCode(400).setStatusMessage("Incorrect email. Must be of type @husky.neu.edu.").end();
+    }
+
+    //if email is valid send email
+    if(processor.validateEmail(email)) {      
+      if(!resetTokenMap.containsKey(email)){
+        String token = createJWT(email, "auth-token", TOKEN_DURATION * 24, 0, true);
+        validToken.put(token, new Date());
```
Comment (rymaju) [inline api/src/main/java/com/codeforcommunity/rest/ApiRouter.java:917]: util.Date

**Review by rymaju — CHANGES_REQUESTED** [review]
Most of this is good, I like how you seperated things into a util class, very clean. I have a couple of other suggestions though. Obviously we need to do the whole util.Date and sql.Date fix, but also we need a route that calls the reset password method that you defined. We just want a simple POST route that calls the method at the top of ApiRouter.java where we initialize all the routes, really it ought to be rate limited too so we dont spam people with emails, but if thats too much work we can seperate it onto a new ticket.

**Comment by VishikBhalla** [comment]
I will add the changes this weekend

**Comment by jackblanc** [comment]
Please provide a status update on this ticket

**Comment by jackblanc** [comment]
Closing this, as no update has been provided and this PR is quite old

**Pull request closed without merging**